### PR TITLE
Support new Lists endpoints

### LINKED ISSUE #572: Support new Lists endpoints
The v2 API is officially launched now and has endpoints for reading Lists, which are important to have: 

https://twittercommunity.com/t/announcing-list-lookup-endpoints-for-the-twitter-api-v2/161965

https://developer.twitter.com/en/docs/twitter-api/lists/list-lookup/introduction

### PR DESCRIPTION
**Pull request by igorbrigadir**
fix #572

API Returning Lists:

- [x] https://developer.twitter.com/en/docs/twitter-api/lists/list-lookup/api-reference/get-lists-id
- [x] https://developer.twitter.com/en/docs/twitter-api/lists/list-lookup/api-reference/get-users-id-owned_lists
- [x] https://developer.twitter.com/en/docs/twitter-api/lists/list-follows/api-reference/get-users-id-followed_lists
- [x] https://developer.twitter.com/en/docs/twitter-api/lists/pinned-lists/api-reference/get-users-id-pinned_lists
- [x] https://developer.twitter.com/en/docs/twitter-api/lists/list-members/api-reference/get-users-id-list_memberships

API Returning Users:

- [x] https://developer.twitter.com/en/docs/twitter-api/lists/list-members/api-reference/get-lists-id-members
- [x] https://developer.twitter.com/en/docs/twitter-api/lists/list-follows/api-reference/get-lists-id-followers

API Returning Tweets:

- [x] https://developer.twitter.com/en/docs/twitter-api/lists/list-tweets/api-reference/get-lists-id-tweets

CLI:

- [x] twarc2 lists all [user_id] 
- [x] twarc2 lists owned [user_id] 
- [x] twarc2 lists followed [user_id] 
- [x] twarc2 lists memberships [user_id]
- [x] twarc2 lists lookup [list_id]
- [x] twarc2 lists bulk-lookup [list_id]
- [x] twarc2 lists followers [list_id]
- [x] twarc2 lists members [list_id]
- [x] twarc2 lists tweets [list_id]

### CODE REVIEW

**Comment by igorbrigadir** [comment]
There's a bit of extra refactoring with `_write_with_progress` function with this too.

**Comment by igorbrigadir** [comment]
Ok, i think i managed to make all the commands work - there are still some inconsistencies with other commands, but i think this PR is large enough as it is, we can address the major inconsistencies in #608

**Comment by SamHames** [comment]
Ooh, exciting - I should have time to look at it this time tomorrow.

File: twarc/client2.py [hunk 1]
```diff
@@ -294,46 +302,323 @@ def _search(
...
+        return self._lists(
+            url=url,
+            expansions=expansions,
+            list_fields=list_fields,
+            user_fields=user_fields,
+            max_results=max_results,
+            pagination_token=pagination_token,
+        )
+
+    def followed_lists(
```
Comment (SamHames) [inline twarc/client2.py:484]: I was going to ask why this was followed (past tense) and not following (present tense) like the other following functionality, but I see that's what Twitter calls it...

**Review by SamHames — APPROVED** [review]
Looks good to me - found some interesting things poking around a few lists and didn't run into any issues with the functionality.

I pushed a few docstring tweaks for things that confused me until I looked at the output data.

File: twarc/client2.py (hunk 1 above)
Comment (igorbrigadir) [reply twarc/client2.py:484]: yeah i thought i'd stick with the API docs for that one

**Comment by igorbrigadir** [comment]
Oh thanks for catching those docstrings!

**Pull request merged**